### DISCUSSION TITLE
OPL PS2 SMB Server Connection Error

### DISCUSSION THREAD

**Discussion by alastairbdoggett** [retronas/discussions#2] [Q&A] [answered]
I have successfully followed the installation instructions in the [OPL wiki](https://github.com/danmons/retronas/wiki/OpenPS2Loader) and have RetroNAS running on a Raspberry Pi 4B.

The slight difference in my set up is I want to use the RPi as a WiFi Ethernet bridge to the PS2. That is to say, have the RPi connected wirelessly to my router and then a short ethernet cable between the PS2 and the RPi.

I did get this working in the past but I have since created fresh install and can not get it working again. I get a `300` error on the SMB share and it makes me think that maybe my Network Settings need to be tweaked or create some kind of wifi->eth route like this [script](https://github.com/toolboc/psx-pi-smbshare/blob/master/wifi-to-eth-route.sh).

Any help would be greatly appreciated and let me know if you need more details!

**Comment by sairuk** [comment]
since this varies from the standard RetroNAS offering of OPL/Samba converted to discussion

**Comment by alastairbdoggett** [comment]
```
root@retronas:~# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host noprefixroute 
       valid_lft forever preferred_lft forever
2: eth0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc mq state DOWN group default qlen 1000
    link/ether dc:a6:32:e8:b9:e7 brd ff:ff:ff:ff:ff:ff
3: wlan0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether dc:a6:32:e8:b9:e8 brd ff:ff:ff:ff:ff:ff
    inet 192.168.1.197/24 brd 192.168.1.255 scope global dynamic noprefixroute wlan0
       valid_lft 258275sec preferred_lft 258275sec
    inet6 2603:8000:a403:62d1::ee8/128 scope global dynamic noprefixroute 
       valid_lft 546722sec preferred_lft 546722sec
    inet6 2603:8000:a403:62d1:2c7d:eb36:99b0:1bf6/64 scope global dynamic noprefixroute 
       valid_lft 547601sec preferred_lft 547601sec
    inet6 fe80::eca1:410d:7ba1:c618/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever
```
![IMG_0589](https://github.com/danmons/retronas/assets/17734670/6ec7c950-2ac9-41d7-88dd-c747383a6ffe)

**Reply by alastairbdoggett** [reply, marked as answer]
These are my Network Settings and from the `ip a` stdout you can see the wifi is not being shared with eth device. 

I ran the [script](https://github.com/toolboc/psx-pi-smbshare/blob/master/wifi-to-eth-route.sh) (the only pre-req I needed was to install `iptables`) which indeed gives back:

```
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether dc:a6:32:e8:b9:e7 brd ff:ff:ff:ff:ff:ff
    inet 192.168.2.1/24 brd 192.168.2.255 scope global eth0
       valid_lft forever preferred_lft forever
```

This then resolved the issue with updated Network Settings:

```
-PS2-
...
IP Address: 192.168.2.2
...
Gateway: 192.168.2.1


-SMB SERVER-
...
IP Address: 192.168.2.1     <<<<<<----- I think this was the the missing piece plugged by the wifi->eth script
...
```

This resolved the issue ✅

**Reply by sairuk** [reply]
Just a note, as we move forward with the firewalld branch (no eta) what this script is doing may come into conflict with what we implement

i currently run two zones modern/retro and forward select traffic eth have an eth and wireless adapter assigned to them. I will try to remember to take a look at this script when i revisit that new branch


you should've just been able to route traffic between retronas/ps2 using the retronas eth address on the ps2 tho

that script has prob just assigned a static address (missing in your original ip output) in a new subnet (from the look of the output) to the eth adapter which allows traffic to route, ps2 should be pointing at .2

you shouldn't need iptables, or ipforwarding etc for just accesing the samba share

**Reply by alastairbdoggett** [reply]
```
you should've just been able to route traffic between retronas/ps2 using the retronas eth address on the ps2 tho
```
I wasn't getting an eth address from retronas before
```
2: eth0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc mq state DOWN group default qlen 1000
    link/ether dc:a6:32:e8:b9:e7 brd ff:ff:ff:ff:ff:ff
```
or am i misinterpreting something?


```
you shouldn't need iptables, or ipforwarding etc for just accesing the samba share
```
could you elaborate a little on how this translates to the Network Settings that i need to provide in OPL?

**Reply by alastairbdoggett** [reply]
ideally i do not want to have to run this script 😅

**Reply by alastairbdoggett** [reply]
somewhat related:  i am seeing inconsistent failures `NetworkManager-wait-online.service: Failed with result 'exit-code'.` on startup (presumably some timeout). 

note that this prevents the aforementioned solution to my problem from working -- i have to manually restart `NetworkManager-wait-online.service` to resolve

**Reply by sairuk** [reply]
retronas is not (yet) responsible for networking on the device it is installed to, even when we do it will be a few pre-canned scenarios and wont be a one size fits all. retronas currently provides the retro services of which only some depend on networking being correctly configured. 

you are correct in that your ethernet device did not have an ip address to begin with. No ip address on eth0 is what was causing you issues accessing the samba share over ethernet. it is my recollection that `raspi-config` can configure static up addressing for interfaces.

Your `rpi<-wireless->router` is separate from your `ps2<-ethernet->lan` connection and should be unless there is a very specific reason you want you PS2 to have internet access through your rpi. There may well be a reason, only you could say.

If an ip address was assigned to eth0 and samba was listening on the same interface then the ps2 should be configured to connect directly to the ethernet ip address not the wireless one which along with some other settings is what that script did.

in the most basic configuration that we implement in retronas the tools iptables, ipforwarding and dnsmasq are not used because we are not responsible for networking nor firewalling nor forwarding at the moment. The implemented samba service simply listens on all available network interface. "available" in this case is that there is an ip address assigned and the connection is up

**Reply by alastairbdoggett** [reply]
Thanks for the details! I don't have much (if any) experience with networking-related issues so this is very helpful. I did a check of `raspi-config` and didn't see anything that allowed me to set a static ip to eth0 (my takeaway was this was the core reason for my issue). Will post back if I am able to get this working without the help of the script, but for now this workaround seems stable.
